### ISSUE TITLE
PassengerAgent segfault 

### ISSUE THREAD

**Issue by iMHLv2** [phusion/passenger#2396]
Please follow the steps below if you have found an **issue with Passenger**, or would like to request a feature.

Post to [Stack Overflow](https://stackoverflow.com/questions/tagged/passenger) instead if you are not sure, or if you have a question about your configuration.

---

## Issue report

Fill in as much as possible so that we can understand, find and fix the problem.

**Are you sure this is a bug in Passenger?**
Please read our troubleshooting and debugging guides first:
https://www.phusionpassenger.com/library/admin/troubleshooting/
https://www.phusionpassenger.com/library/admin/

**Please try with the newest version of Passenger to avoid issues that have already been fixed**

**Question 1: What is the problem?**
- What is the expected behavior?
- What is the actual behavior?
- How can we reproduce it? Please try to provide a sample application (or Virtual Machine) demonstrating the issue. Otherwise, if we can't reproduce it, we might have to ask you a number of followup questions or run certain commands to try and figure out the problem.

Be as detailed as possible in your descriptions, include any logs and stack traces (don't just cut/paste the error, provide some logging before that too).

(if you are requesting a feature instead of reporting an issue, describe here what you have in mind and how it would help you)

Your answer:

Unfortunately, not a lot of details are available. Every couple of weeks, nginx starts returning HTTP 500s for any Node APIs running via Passenger. Static files served by nginx are unaffected and continue to work fine. We aren't specifying passenger_log_file and there are no relevant errors in /var/log/nginx/error.log. The only indication (other than 500s) is in /var/log/messages:

The first two occurred while using Passenger 6.0.6:

```
Oct 12 18:37:19 $HOSTNAME kernel: PassengerAgent[28352]: segfault at 0 ip 0000000000491783 sp 00007f1ac4de8540 error 4 in PassengerAgent[400000+376000]

Oct 22 10:27:20 $HOSTNAME kernel: PassengerAgent[2416]: segfault at 0 ip 0000000000491783 sp 00007f0a437f6540 error 4 in PassengerAgent[400000+376000]
```

We upgraded to Passenger 6.0.11 and it happened again:

```
Nov 16 08:49:57 $HOSTNAME kernel: PassengerAgent[28389]: segfault at 0 ip 000000000049589f sp 00007fc286ff5540 error 4 in PassengerAgent[400000+378000]
```

**Question 2: Passenger version and integration mode**:
- For example: open source 5.0.26 standalone; enterprise 5.0.21/nginx

Your answer:

We have experienced the issue with the following versions (both open source):

* nginx/1.16.1 Phusion_Passenger/6.0.6 (nginx-mod-http-passenger-6.0.6-1.el7.x86_64 via yum)
* nginx/1.20.1 Phusion_Passenger/6.0.11 (nginx-mod-http-passenger-6.0.11-1.el7.x86_64 via yum)

**Question 3: OS or Linux distro, platform (including version)**:
- For example: Debian 8, x86_64 or OS X 10.10 Yosemite, x86_64

Your answer:

* Red Hat Enterprise Linux Server 7.9 (Maipo)
* /proc/version: 

```
Linux version 3.10.0-1160.31.1.el7.x86_64 (mockbuild@x86-040.build.eng.bos.redhat.com) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) ) #1 SMP Wed May 26 20:18:08 UTC 2021
```

**Question 4: Passenger installation method**:

Your answer:
[ ] RubyGems + Gemfile
[ ] RubyGems, no Gemfile
[ ] Phusion APT repo
[x] Phusion YUM repo
[ ] OS X Homebrew
[ ] source tarball
[ ] Other, please specify:

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:
- For example: Ruby 2.3.0, RVM, Rails 5.0.0; Node.js 4.0.0 with Express 4.13.4

Your answer:

* Node.js v12.18.2

**Question 6: Are you using a PaaS and/or containerization? If so which one?**
- For example: Heroku, Amazon Container Services, Docker 1.9 with an image based on passenger-docker

Your answer:

No

**Question 7: Anything else about your setup that we should know?**

Your answer:

We have been using the default passenger_log_level 3, but just bumped it to 5. However, I'm not optimistic that we'll learn anything, since the segfault seems to kill Passenger before it gets a chance to gracefully clean up or produce logs.

---

We strive for quality and appreciate you taking the time to submit a report! Please note that if you want guaranteed response times and priority issue support we encourage you to join our [enterprise customer](https://www.phusionpassenger.com/features#premium-features) base. They also provide us with the means to continue our high level of open source support!

**Comment by CamJN** [comment]
You're right that's not enough to go on. You might get more info by installing the `passenger-debuginfo` package, but it's not a sure thing.
